tests/test-update-from-main.c

```diff
@@ -47,6 +47,15 @@ test_update_from_main (EosUpdaterFixture *fixture,
   DownloadSource main_source = DOWNLOAD_MAIN;
   g_autoptr(GVariant) main_source_variant = NULL;
 
+  /* We could get OSTree working by setting OSTREE_BOOTID, but shortly
+   * afterwards we hit unsupported syscalls in qemu-user when running in an
+   * ARM chroot (for example), so just bail. */
+  if (!eos_test_has_ostree_boot_id ())
+    {
+      g_test_skip ("OSTree will not work without a boot ID");
+      return;
+    }
```

Comment:
I wonder if we can use a GTestFixture to implement this for all tests as opposed to just some of them (mental note that I need to do this in my own branch too)

Reply:
We can’t, but we don’t want to. I want as many tests as possible to run in an ARM chroot, and to only skip the ones which will spuriously fail.

We can’t use `g_test_skip()` from a fixture setup function, but that’s something which would ideally be [fixed in GLib](https://bugzilla.gnome.org/show_bug.cgi?id=788419).

Reply:
Ah, I see. Yes, it would be nice if that could be fixed in GLib, indeed.